build.sbt

```diff
@@ -122,82 +104,45 @@ def jarsAndListGenerator = Def.taskDyn {
-    val benchGroupContent = new StringBuilder
-    val benchDetailsFile = (resourceManaged in Compile).value / "benchmark-details.properties"
-    val benchDetailsStream = new java.io.FileOutputStream(benchDetailsFile)
     val benchDetails = new java.util.Properties
 
-    // Add the benchmarks from the different project groups.
     for ((project, allJars, loadedJars) <- groupJars) {
-      val jarLine = loadedJars
-        .map(jar => project + "/" + jar.getName)
-        .mkString(",")
+      val jarLine = loadedJars.map(jar => project + "/" + jar.getName).mkString(",")
       val projectShort = project.stripPrefix("benchmarks/")
       jarListContent.append(projectShort).append("=").append(jarLine).append("\n")
-      for ((name, distroLicense, licenses, description, repetitions) <- listBenchmarks(
-             project,
-             allJars
-           )) {
-        if (!nonGpl || distroLicense == "MIT") {
-          benchGroupContent.append(name).append("=").append(projectShort).append("\n")
-          benchDetails.setProperty("benchmark." + name + ".description", description)
-          benchDetails.setProperty("benchmark." + name + ".repetitions", repetitions.toString)
-          benchDetails.setProperty("benchmark." + name + ".distro", distroLicense.toString)
-          benchDetails.setProperty("benchmark." + name + ".licenses", licenses.toString)
+
+      // Scan project jars for benchmarks and fill the property file.
+      for (benchInfo <- listBenchmarks(project, allJars)) {
+        if (!nonGpl || benchInfo.distro() == License.MIT) {
+          for ((k, v) <- benchInfo.toMap()) {
+            benchDetails.setProperty(s"benchmark.${benchInfo.name}.$k", v);
+          }
         }
       }
     }
+
+    val jarListFile = (resourceManaged in Compile).value / "groups-jars.txt"
     IO.write(jarListFile, jarListContent.toString, StandardCharsets.UTF_8)
-    IO.write(benchGroupFile, benchGroupContent.toString, StandardCharsets.UTF_8)
+
+    val benchDetailsFile = (resourceManaged in Compile).value / "benchmark-details.properties"
+    val benchDetailsStream = new java.io.FileOutputStream(benchDetailsFile)
     benchDetails.store(benchDetailsStream, "Benchmark details")
-    benchGroupFile +: benchDetailsFile +: jarListFile +: groupJars.flatMap {
+
+    benchDetailsFile +: jarListFile +: groupJars.flatMap {
       case (_, jars, _) => jars
     }
   }
 }
 
-val renaissanceFormat = taskKey[Unit](
-  "Reformat source code with scalafmt."
-)
-
-def createRenaissanceFormatTask = Def.taskDyn {
-  val formatTasks = for (p <- subProjects) yield scalafmt in (p, Compile)
-  val testFormatTasks = for (p <- subProjects) yield scalafmt in (p, Test)
-  val buildFormatTasks = for (p <- subProjects) yield scalafmtSbt in (p, Compile)
-  flattenTasks(formatTasks ++ testFormatTasks ++ buildFormatTasks)
-}
-
-val renaissanceFormatTask = renaissanceFormat := createRenaissanceFormatTask.value
+addCommandAlias("renaissanceFormat", ";scalafmt;scalafmtSbt")
```

Comment:
Did you check that this alias runs `scalafmt` in all subprojects, and all configurations (e.g. `test` configuration), and in the `sbt` project files?

Reply:
From what I tested (just running it in sbt shell), it does -- the original method (the one that generated tasks) way actually did not run scalafmt for meta projects.

The various configurations of SBT are still confusing to me (as did not have the need for them), so I will keep trying to simplify the build files. For me SBT is currently a tool where the documentation is basically useless (as if a language theoretician wrote it, enjoying the myriad concepts, but never really explaining how things work for some more complex use cases, such as, apparently, ours), and getting something done requires trial and error and StackOverflow. I hate when I don't know exactly what I'm doing, but I don't have time to study SBT completely.

I spent hours trying to do simple things, but the problem is that when something doesn't work as expected, I don't know why. To me the aggregation of commands should be a basic thing for a multi-project setup, yet if you try to google it, there are no satisfying answers. None of them actually did what we had there originally (which did not work 100% anyway) and mostly suggested creating an alias.